Feat/stats leaderboard

shan-226 opened the pull request:
When you click on the leader board stats on the profile it will take you to my page. 

If you want to test the pie chart data:
-Create a positions array for your user in the profile_data collection and populate it with 5 numbers (1st, 2nd, 3rd, top5, top10).

If you want to test the onclick of the pie chart segment:
-Create a document in the leaderboard collection and link it to a valid event and have your userID be one of the fields in the positions array.

The design of the page may change next week when we are redesigning.  

@MonikaTheiss and @LeratoKgomo please review.

shan-226 commented:
@pieryvdl can you please check the GitGuardian check for me and advise me on how to move forward.

pieryvdl commented:
> @pieryvdl can you please check the GitGuardian check for me and advise me on how to move forward.

Don't stress about it, they are all on the test files. It it part of the password validation and not secrets at all.

LeratoKgomo commented:
looks great and works as intended maybe in the future we can handle a case where the user hasn't played any games yet but otherwise im happy , thank you

MonikaTheiss commented:
@shan-226 would it perhaps be possible to add the necessary data for my profile please? I'm worried I might do it wrong, as I'm a bit confused on how and where you did things?

MonikaTheiss commented on lib/view/pages/stats/stats_games.dart:
```diff
@@ -0,0 +1,127 @@
+// import 'package:logger/logger.dart'; // Add this line for logging
+
+// final logger = Logger(); // Initialize the logger
+
```
out of interests sake, what did you use the logger for?

shan-226 replied:
I used it to figure out why my code wasn't working. So I followed the code and added logging until I found the problem.

MonikaTheiss replied:
is it like print? What is different between print and logger?

shan-226 replied:
It's similar to print. The output in the terminal is just highlighted and formatted so that its easier to see what's happening.

MonikaTheiss replied:
ah okay interesting, thank you

MonikaTheiss reviewed (commented):
Code looks good. I dont have the leaderboard data setup on my account, so I cant test the running code just yet. Please let me know if you will be able to add the necessary data to the database for my profile ( starts with oD6) ? I will wait for it and then approve the pull request
A small thing, could you perhaps remove the commented out code in the future please? Just so we don't get confused in the future 😅

shan-226 commented:
> @shan-226 would it perhaps be possible to add the necessary data for my profile please? I'm worried I might do it wrong, as I'm a bit confused on how and where you did things?

I added the data for your profile. I added it to the queenzy profile, I hope its the right one. If you click on the 1st pie chart segment it will show the tournament game. I didn't add the other positions' data to the database so they won't show any data when clicked on.

MonikaTheiss commented:
> nzy pro

perfect I can see it, thanks!

MonikaTheiss reviewed (approved):
Looks great on my side, thank  you !
